integration_test.go

```diff
@@ -1539,46 +1539,38 @@ func TestDeadlockConsumerIssue48(t *testing.T) {
 
 // https://github.com/streadway/amqp/issues/46
 func TestRepeatedChannelExceptionWithPublishAndMaxProcsIssue46(t *testing.T) {
-	var conn *Connection = nil
+	conn := integrationConnection(t, "issue46")
+	if conn == nil {
+		t.Fatal("conn is nil")
+	}
 
 	t.Cleanup(func() {
-		if conn != nil {
-			conn.Close()
-		}
+		conn.Close()
 	})
 
 	for i := 0; i < 100; i++ {
-		if conn == nil || conn.IsClosed() {
-			conn = integrationConnection(t, "issue46")
-			if conn == nil {
-				t.Fatal("conn is nil")
-			}
+		if conn.IsClosed() {
+			t.Fatal("conn is closed")
 		}
 
-		ch, err := conn.Channel()
-		if err, ok := err.(Error); ok {
-			if err.Code != 504 {
-				t.Fatalf("expected channel only exception i: %d got: %+v", i, err)
-			}
+		ch, channelOpenError := conn.Channel()
+		if channelOpenError != nil {
+			t.Fatalf("error opening channel: %d error: %+v", i, channelOpenError)
 		}
 
-		if ch == nil {
-			continue
-		}
-
-		for j := 0; j < 10; j++ {
+		for j := 0; j < 100; j++ {
 			if ch.IsClosed() {
+				if j == 0 {
+					t.Fatal("channel should not be closed")
+				}
 				break
-			} else {
-				err = ch.Publish("not-existing-exchange", "some-key", false, false, Publishing{Body: []byte("some-data")})
-				if err, ok := err.(Error); ok {
-					if err.Code != 504 {
-						t.Fatalf("expected channel only exception i: %d j: %d got: %+v", i, j, err)
-					}
-					if cerr := ch.Close(); cerr != nil {
-						t.Logf("error on channel close i: %d j: %d got: %+v", i, j, cerr)
-					}
-					break
+			}
+			publishError := ch.Publish("not-existing-exchange", "some-key", false, false, Publishing{Body: []byte("some-data")})
+			if publishError, ok := publishError.(Error); ok {
```

Comment:
```
if publishError, ok := publishError.(Error); ok {
```
is always false since the error is a pointer

it should be:
```
if publishError, ok := publishError.(*Error); ok {
```

or maybe use the `switch` case to test the case

Reply:
Ah! Great catch.